src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonParser.JsonNetSyntaxChecks.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.Json
+{
+    using static EmbeddedSyntaxHelpers;
+
+    using JsonToken = EmbeddedSyntaxToken<JsonKind>;
+
+    internal partial struct JsonParser
+    {
+        private partial struct JsonNetSyntaxChecker
```

Comment:
Why partial?

Reply:
fixed.  #resolved.

Reply:
Is there a spec or doc we could point to for both sets of rules (strict and Json.Net)?

Reply:
Can we make all the methods here static? (same for the other checker)

Reply:
done.  also made them into static classes.